ui-components/src/page_components/sign-in/FormSignInMFAType.tsx

```diff
@@ -0,0 +1,111 @@
+import React from "react"
+import {
+  AppearanceStyleType,
+  Button,
+  Field,
+  Form,
+  FormCard,
+  Icon,
+  t,
+  AlertBox,
+  SiteAlert,
+  AlertNotice,
+  ErrorMessage,
+} from "@bloom-housing/ui-components"
+import type { UseFormMethods } from "react-hook-form"
+import { FormSignInNetworkError } from "./FormSignIn"
+import { EnumRequestMfaCodeMfaType } from "@bloom-housing/backend-core/types"
+
+export type FormSignInMFAProps = {
+  control: FormSignInMFAControl
+  onSubmit: (data: FormSignInMFAValues) => void
+  networkError: FormSignInNetworkError
+}
+
+export type FormSignInMFAControl = {
+  errors: UseFormMethods["errors"]
+  handleSubmit: UseFormMethods["handleSubmit"]
+  register: UseFormMethods["register"]
+  setValue: UseFormMethods["setValue"]
+}
+
+export type FormSignInMFAValues = {
+  mfaType: EnumRequestMfaCodeMfaType
+}
+
+const FormSignInMFAType = ({
+  onSubmit,
+  networkError,
+  control: { errors, register, handleSubmit, setValue },
+}: FormSignInMFAProps) => {
+  const onError = () => {
+    window.scrollTo(0, 0)
+  }
+
+  return (
+    <FormCard>
+      <div className="form-card__lead text-center border-b mx-0">
+        <Icon size="2xl" symbol="profile" />
+        <h2 className="form-card__title">{t("nav.signInMFA.verificationChoiceMainTitle")}</h2>
+        <p className="form-card__sub-title">
+          {t("nav.signInMFA.verificationChoiceSecondaryTitle")}
+        </p>
+      </div>
+      {Object.entries(errors).length > 0 && !networkError.error && (
+        <AlertBox type="alert" inverted closeable>
+          {errors.authentication ? errors.authentication.message : t("errors.errorsToResolve")}
+        </AlertBox>
+      )}
+
+      {!!networkError.error && Object.entries(errors).length === 0 && (
+        <ErrorMessage id={"householdsize-error"} error={!!networkError.error}>
+          <AlertBox type="alert" inverted onClose={() => networkError.reset()}>
+            {networkError.error.title}
+          </AlertBox>
+
+          <AlertNotice title="" type="alert" inverted>
+            {networkError.error.content}
+          </AlertNotice>
+        </ErrorMessage>
+      )}
```

Comment:
This section is repeated three different times, can you split this off into its own component, which can be unit tested.